otelx/semconv/context.go

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package semconv
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/urfave/negroni"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/ory/x/httpx"
+)
+
+type contextKey int
+
+const contextKeyAttributes contextKey = iota
+
+func ContextWithAttributes(ctx context.Context, attrs ...attribute.KeyValue) context.Context {
+	existing, _ := ctx.Value(contextKeyAttributes).([]attribute.KeyValue)
```

Comment:
Is it safe to ignore the second value here? What happens if the type assertion fails?

Reply:
We use the `a, _ := ` syntax here to prevent the type assertion from panicking if the key does not exist in the context. Instead of panicking, the assertion will return `nil, false`. But we don't care about the `false`, since we proceed in the same manner anyway, i.e., appending to a (possibly nil) slice in the next step.

Reply:
I see, thanks for the clarification.